### PR TITLE
Add `--version` flag

### PR DESCRIPTION
**Pull request by KnorpelSenf**
Implements `--version` and `-v` for the CLI. Example output:

```bash
$ deno2node --version
deno2node 1.0.1
typescript 4.4.4
```

### CODE REVIEW

**Comment by KnorpelSenf** [comment]
This requires the `ts-morph` version string to be updated manually whenever a the ts-morph dependency is updated.

Note also that future version released must be performed using `npm version` so that the correct version string of `deno2node` can be injected before the release.

File: package.json [hunk 1]
```diff
@@ -21,6 +21,8 @@
     "lint": "npm run --silent clean && deno lint --ignore=node_modules/",
     "test": "npm run --silent prepare",
     "prepare": "deno run --allow-read --allow-write='lib/' --allow-env src/cli.ts src/tsconfig.json",
+    "inject-version": "echo \"export const version = \\\"$npm_package_version\\\";\" > src/version.ts",
+    "version": "npm run inject-version && git add src/version.ts && git commit -m \"Update version to $npm_package_version\"",
```
Comment (wojpawlik) [inline package.json:25]: I dislike this. `deps.deno.ts` could use `fetch` relative to `import.main.meta`.

File: src/deps.node.ts [hunk 2]
```diff
@@ -3,3 +3,4 @@
 export { default as path } from "path";
 export { z } from "zod";
 export * from "ts-morph";
+export const tsMorphVersion = "12.2.0";
```
Comment (wojpawlik) [inline src/deps.node.ts:6]: Unnecessary. `ts.version` is sufficient and easy to obtain.

**Review by wojpawlik — CHANGES_REQUESTED** [review]

File: src/deps.node.ts (hunk 2 above)
Comment (wojpawlik) [reply src/deps.node.ts:6]: Alternatively, perhaps it could be read from `package*.json`

Comment (wojpawlik) [reply src/deps.node.ts:6]: And it might change across `npm install`s, since npm allows semver-minor updates by default

File: src/deps.node.ts [hunk 3]
```diff
@@ -3,3 +3,8 @@
 export { default as path } from "path";
 export { z } from "zod";
 export * from "ts-morph";
+
+export function getOwnVersion(): string {
+  const packageJson = require("./package.json");
```
Comment (wojpawlik) [inline src/deps.node.ts:8]: Won't work, `deno2node` is ESM.

File: src/deps.deno.ts [hunk 4]
```diff
@@ -1,4 +1,14 @@
...
-export * as path from "https://deno.land/std@0.106.0/path/mod.ts";
-export * from "https://deno.land/x/ts_morph@12.0.0/mod.ts";
+export { z } from "https://cdn.skypack.dev/zod@v3.11.6?dts";
+export * as path from "https://deno.land/std@0.115.0/path/mod.ts";
+export * from "https://deno.land/x/ts_morph@12.2.0/mod.ts";
+import { dirname, join } from "https://deno.land/std@0.115.0/path/mod.ts";
+
+export async function getOwnVersion(): Promise<string> {
+  const thisDir = dirname(import.meta.url);
+  const path = join(thisDir, "..", "package.json");
```
Comment (wojpawlik) [inline src/deps.deno.ts:10]: Slight simplification:

```suggestion
  const path = join(import.meta.url, "../..", "package.json");
```

**Review by wojpawlik — CHANGES_REQUESTED** [review]

File: src/deps.node.ts (hunk 3 above)
Comment (KnorpelSenf) [reply src/deps.node.ts:8]: Good point …

File: src/shim.node.ts [hunk 5]
```diff
@@ -10,3 +10,14 @@ export const Deno = {
...
 };
+
+export async function fetch(
+  localPath: string,
+  // deno-lint-ignore no-explicit-any
+): Promise<{ json: () => Promise<any> }> {
+  if (!localPath.startsWith("file:")) {
+    throw new Error("Can only read local files!");
+  }
+  const data = await readFile(localPath.substr(5), { encoding: "utf-8" });
```
Comment (wojpawlik) [inline src/shim.node.ts:21]: Slighly neater with `URL`:

```suggestion
  url: URL,
  // deno-lint-ignore no-explicit-any
): Promise<{ json: () => Promise<any> }> {
  if (url.protocol !== "file:") {
    throw new Error("Can only read local files!");
  }
  const data = await readFile(url, { encoding: "utf-8" });
```

File: src/deps.deno.ts (hunk 4 above)
Comment (KnorpelSenf) [reply src/deps.deno.ts]: Must use `dirname` on Deno.

File: src/cli.ts [hunk 6]
```diff
@@ -1,12 +1,28 @@
...
   Deno.exit(2);
 }
 
+if (Deno.args.length !== 1) {
+  printUsageAndExit();
+} else if (Deno.args[0].startsWith("-")) {
+  if (Deno.args[0] === "-v" || Deno.args[0] === "--version") {
+    const thisDir = path.dirname(import.meta.url);
+    const file = path.join(thisDir, "..", "package.json");
+    const res = await fetch(file);
```
Comment (wojpawlik) [inline src/cli.ts:16]:
```suggestion
    const url = new URL("../../package.json", import.meta.url);
    const res = await fetch(url);
```

**Review by wojpawlik — APPROVED** [review]

**Comment by KnorpelSenf** [comment]
How would we add tests for this?

**Comment by KnorpelSenf** [comment]
That was simple. Thanks!

**Pull request merged**